ts/session/types/OpenGroup.ts

```diff
@@ -0,0 +1,82 @@
+// This is the Open Group equivalent to the PubKey type.
+
+interface OpenGroupParams {
+  server: string;
+  channel: number;
+  conversationId: string;
+}
+
+export class OpenGroup {
+  private static readonly serverRegex = new RegExp(
+    '^([\\w-]{2,}.){1,2}[\\w-]{2,}$'
+  );
+  private static readonly groupIdRegex = new RegExp(
+    '^publicChat:[0-9]*@([\\w-]{2,}.){1,2}[\\w-]{2,}$'
+  );
+  public readonly server: string;
+  public readonly channel: number;
+  public readonly groupId?: string;
+  public readonly conversationId: string;
+
+  constructor(params: OpenGroupParams) {
+    const strippedServer = params.server.replace('https://', '');
+    this.server = strippedServer;
+
+    // Validate server format
+    const isValid = OpenGroup.serverRegex.test(this.server);
+    if (!isValid) {
+      throw Error('an invalid server or groupId was provided');
+    }
+
+    this.channel = params.channel;
+    this.conversationId = params.conversationId;
+    this.groupId = OpenGroup.getGroupId(this.server, this.channel);
+  }
+
+  public static from(
+    groupId: string,
+    conversationId: string
+  ): OpenGroup | undefined {
+    // Returns a new instance from a groupId if it's valid
+    // eg. groupId = 'publicChat:1@chat.getsession.org'
+
+    const server = this.getServer(groupId);
+    const channel = this.getChannel(groupId);
+
+    // Was groupId successfully utilized?
+    if (!server || !channel) {
+      return;
+    }
+
+    const openGroupParams = {
+      server,
+      channel,
+      groupId,
+      conversationId,
+    } as OpenGroupParams;
+
+    if (this.serverRegex.test(server)) {
+      return new OpenGroup(openGroupParams);
+    }
+
+    return;
```

Comment:
I would make it the oter way:
You test for case of returning undefined since the start of the method.
Also the last return is not needed in your current state

```suggestion
    if (!this.serverRegex.test(server)) {
      return;
    }

    return new OpenGroup(openGroupParams);
```

Reply:
Fixed.